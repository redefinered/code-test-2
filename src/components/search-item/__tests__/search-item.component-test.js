import React from 'react';
import SearchItem from '../search-item.component';

import renderer from 'react-test-renderer';

import mockresults from '../__mocks__/results-mock';

const ITEM = mockresults.results.find(
  item => typeof item.trackName !== 'undefined',
);

const ITEM_WITH_UNDEFINED_TRACK_NAME = mockresults.results.find(
  item => item.collectionId === '379168762',
);

describe('search-item tests', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<SearchItem item={ITEM} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correct if trackName is is undefined', () => {
    const tree = renderer
      .create(<SearchItem item={ITEM_WITH_UNDEFINED_TRACK_NAME} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
