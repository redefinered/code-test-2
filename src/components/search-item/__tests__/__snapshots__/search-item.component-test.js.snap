// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-item tests renders correct if trackName is is undefined 1`] = `<View />`;

exports[`search-item tests renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Enter Sandman
  </Text>
  <Text>
    Metallica
  </Text>
</View>
`;
